design/one-pager-package-runtime-config.md

```diff
@@ -0,0 +1,222 @@
+  requires an `apiVersion` and `kind`.
+
+If the `runtimeConfigRef` is omitted it will default to a runtime config named
+"default" of the type specified by the `--package-runtime` flag. For example
+when Crossplane is run with `--package-runtime=Deployment` a Provider will use a
+DeploymentRuntimeConfig. This behavior matches that of ProviderConfigs. If you
+omit the `providerConfigRef` when creating an MR Crossplane defaults to using a
+ProviderConfig named default. 
+
+Automatically setting a default runtime config has two advantages:
+
+* The default configuration is no longer [hardcoded][hardcoded-pkg-deployment]
+  into the package manager.
+* Administrators can override the configuration all runtimes use by default.
+
+The Crossplane init container will create a default DeploymentRuntimeConfig at
+install time. A Crossplane administrator could then replace it with their own.
+For example Crossplane might install a default DeploymentRuntimeConfig that
+limits all package runtimes to 1 CPU core, but a Crossplane administrator might
+wish to change this to give all runtimes 2 CPU cores by default. Individual
+packages can still be explicitly configured to use a specific runtime config.
+
+Given that we saw ControllerConfig growing into a template for a Deployment, I
+propose we lean into that and make DeploymentRuntimeConfig exactly that. For
+example:
+
+```yaml
+apiVersion: pkg.crossplane.io/v1alpha1
+kind: DeploymentRuntimeConfig
+metadata:
+  name: default
+spec:
+  deploymentTemplate:
+    metadata:
+      labels:
+        example: label
+    spec:
+      replicas: 1
+      template:
+        securityContext:
+          runAsNonRoot: true
+          runAsUser: 2000
+          runAsGroup: 2000
+        containers:
+          # The container used to run the Provider or Function must be named
+          # 'package-runtime'. The package manager will overlay the package's
+          # runtime image, pull policy, etc into this container.
+        - name: package-runtime
+          securityContext:
+            runAsNonRoot: true
+            runAsUser: 2000
+            runAsGroup: 2000
+            privileged: false
+            allowPrivilegeEscalation: false
+          ports:
+          - name: metrics
+            containerPort: 8080
+    # A DeploymentRuntimeConfig can also be used to configure the Service and
+    # ServiceAccount the package manager creates to support the Deployment.
+    serviceTemplate: {}
+    serviceAccountTemplate: {}
+```
+
+The above DeploymentRuntimeConfig matches the values that are currently
+[hardcoded into the package manager][hardcoded-pkg-deployment]. The
+`deploymentTemplate`, etc fields are similar to a Deployment's `template` field.
+
+The package manager will always be opinionated about some things, and will
+overlay the following settings over the top of the provided template:
+
+* The name and namespace of the deployment, service, and service account.
```

Comment:
Actually, I think we might need to let folks configure these names. Or at least the service account name to avoid a regression of https://github.com/crossplane/crossplane/issues/2295.